File: Data Structures & Algorithms/valid-sudoku/submission-0.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Map<Integer,Set> column = new HashMap<>();
        Map<Integer,Set> row = new HashMap<>();
        Map<List<Integer>,Set> squares = new HashMap<>();

        for(int i=0;i<9;i++){
            column.put(i,new HashSet<>());
            row.put(i,new HashSet<>());
        }

        for(int i=0;i<3;i++){
            for(int j=0;j<3;j++){
                squares.put(Arrays.asList(i,j), new HashSet<>());
            }
        }

        for(int i=0;i<board.length;i++){
            for(int j=0;j<board[0].length;j++){
                if(board[i][j]=='.') continue;
                int d = board[i][j] - '0';
                if(!column.get(i).add(d)) return false;
                if(!row.get(j).add(d)) return false;
                if(!squares.get(Arrays.asList(i/3,j/3)).add(d)) return false;
            }
        }

        return true;
    }
}

File: Data Structures & Algorithms/islands-and-treasure/submission-0.cpp
class Solution {
public:
    int dr[4] = {-1,1,0,0};
    int dc[4] = {0,0,-1,1};
    void islandsAndTreasure(vector<vector<int>>& grid) {
        if(grid.size()==0) return;
        for(int i=0;i<grid.size();i++){
            for(int j=0;j<grid[0].size();j++){
                if(grid[i][j]==0){
                    dfs(grid,i,j);
                }
            }
        }
        return;
    }

    void dfs(vector<vector<int>>& grid, int x, int y){
        int distance = grid[x][y] + 1;
        for(int i=0;i<4;i++){
            int row = x + dr[i];
            int column = y + dc[i];
            if(row>=0 && row<grid.size() && column>=0 && column<grid[0].size() && grid[row][column] > distance){
                grid[row][column] = distance;
                dfs(grid,row,column);
            }
        }
    }
};
